src/buffer/out/textBuffer.cpp

```diff
@@ -1832,6 +1859,30 @@ const TextBuffer::TextAndColor TextBuffer::GetText(const bool includeCRLF,
     return data;
 }
 
+size_t TextBuffer::SpanLength(const til::point coordStart, const til::point coordEnd) const
+{
+    const auto bufferSize = GetSize();
+    // The coords are inclusive, so to get the (inclusive) length we add 1.
+    const auto length = bufferSize.CompareInBounds(coordEnd, coordStart) + 1;
+    return gsl::narrow<size_t>(length);
+}
+
+// Routine Description:
+// - Retrieves the plain text data between the specified coordinates.
+// Arguments:
+// - trimTrailingWhitespace - remove the trailing whitespace at the end of the result.
+// - start - where to start getting text (should be at or prior to "end")
+// - end - where to end getting text
+// Return Value:
+// - Just the text.
+std::wstring TextBuffer::GetPlainText(const til::point& start, const til::point& end) const
+{
+    const auto& buffer = _activeBuffer();
+    const auto rects = buffer.GetTextRects(start, end, false, true);
+    const auto textData = buffer.GetText(false, false, rects, nullptr, false);
+    return textData.text;

```

Comment:
@carlos-zamora I wonder if you accidentally committed something in an intermediate state--this bit does not compile. I thought it would be as simple as getting rid of the first line (the one that calls `_activeBuffer()`, because `this` is the buffer) (and replace the instances of `buffer` with `this`), but there's another problem on top of that: `textData.text` yields a vector of strings, but this code was intended to return a single string for the span (and that's what the caller needs). So maybe you intended to join the strings?

But actually what is breaking match-all mode for a single, non-block selection that spans more than one line is the change to `GetTextSpans`. The whole point of the original code there was that for a non-block selection, it should only return a single span.

Reply:
Let me demonstrate the scenario I am talking about, which is now broken. Imagine you have selected the words "terminal rocks" where the text was wrapped across a line:

```
  ┌──────────────┐
  │              │
  │     terminal │
  │rocks         │
  │              │
  └──────────────┘
```

Then (with the `enableColorSelection` feature turned on), you type <kbd>alt</kbd>+<kbd>shift</kbd>+<kbd>6</kbd>, to color all instance of "terminal rocks" red.

With the code as Carlos committed in cf662898a0decb04baad6ef7a8a3c0088fb5aecd, all instances of "terminal" will be colored red (whether or not it is followed by "rocks"), and all instances of "rocks" will be colored red (whether or not it is preceded by "terminal"). But only the full string "terminal rocks" should be colored, not whatever random pieces happened to be split across lines.

Reply:
Oof. I'm sorry. Ok so then if we join the strings to form a single span of text, it should work as expected?

My bad. I definitely got ahead of myself.

Reply:
> if we join the strings to form a single span of text, it should work as expected?

No: `GetTextSpans` also needs to be fixed. The way you have put it, a single, non-blockmode selection that crosses multiple lines will return multiple spans. But that doesn't make sense for what we need here: if I select the word "this", it needs to be a single span, whether or not the "th" shows up on the end of one line and the "is" shows up at the beginning of the next line, or not. Does that make sense?

Reply:
Ok. I get the issue now. And now I understand that we shouldn't use `GetTextRects` because it does a lot of extra work breaking up the selected text by row, when `GetTextSpans` (and `GetPlainText`) don't need that at all. Completely undid my changes to this file and added a few `const` that I saw were missing.

Really sorry about that, again.

Reply:
No problem. I confirm it builds and works now--thanks!